src/osemgrep/networking/Unit_Login.ml

```diff
@@ -0,0 +1,151 @@
+(* Austin Theriault
+ *
+ * Copyright (C) 2019-2023 Semgrep, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+open Testutil
+
+(*****************************************************************************)
+(* Helpers *)
+(*****************************************************************************)
+
+let ok_token = "ok_token"
+let bad_token = "bad_token"
+
+let login_session =
+  ( Uuidm.of_string "00000000-0000-0000-0000-000000000000" |> Option.get,
+    snd (Semgrep_login.make_login_url ()) )
+
+let fake_settings =
+  "has_shown_metrics_notification: true\n\
+   anonymous_user_id: 5a06eb03-071d-4c89-b297-993c4912fa8f"
+
+let with_login_client =
+  let make_fn req body =
+    let uri = Cohttp.Request.uri req in
+    match Uri.path uri with
+    | "/api/agent/deployments/current" ->
+        let status, body =
+          match Testing_client.get_header req "Authorization" with
+          | Some "Bearer ok_token" -> (200, "./tests/login/ok_response.json")
```

Comment:
Maybe it would be better to return directly the content of the file instead of the file?
So we can avoid creating those tests/ files.
Or maybe change Testing_client body to be a variant such as File of Fpath.t | Content of string?

Reply:
Yea @kopecs and I discussed this. I believe he's going to change that soon, I didn't want to step on his toes

Reply:
Yeah see #8667